data/data.go

```diff
@@ -18,6 +24,78 @@ type Rule struct {
 	Condition  string        `json:"condition"`
 }
 
+// Serialize builds the Yara rule
+func (rule *Rule) Serialize(output io.Writer) {
```

Comment:
I think ultimately, these serialization functions can go in a `utils.go` file.  They will still be in the same package (directory); just a different file.  We can modify it later, though, and it won't affect the library, which is great.

Also, I think maybe what we can do later on is use TextMarshaler on each of the types so that a ruleset can recursively be serialized.  I need to do more research on it, though; I'm not 100% sure it's the best tool for the job.